lmnet/executor/train.py

```diff
@@ -40,7 +41,12 @@ def setup_dataset(config, subset, rank):
     dataset_kwargs = dict((key.lower(), val) for key, val in config.DATASET.items())
     dataset = DatasetClass(subset=subset, **dataset_kwargs)
     enable_prefetch = dataset_kwargs.pop("enable_prefetch", False)
-    return DatasetIterator(dataset, seed=rank, enable_prefetch=enable_prefetch)
+    batch_pre_fetch = dataset_kwargs.get('batch_pre_fetch', 3)      # Default is arbitrarily set to 3, can be modified based on availabel RAM
```

Comment:
The default value setting of `batch_pre_fetch` and `num_parallel_process` should be in `DatasetIterator`, because if the default value setting is outside of class, It takes cost to read and understand the code. Plz change 👍

Reply:
@ruimashita San:
> No need for `batch_pre_fetch` value as queue is the batch size data.

To set `num_parallel_process` default value in `DatasetIterator` I've added `num_parallel_process` attribute to the `Base` class of the dataset class. Through dataset class attribute `num_parallel_process` config variable can be read and set in class initializer `DatasetIterator.__init__`.